proposed/event-dispatcher-meta.md

```diff
@@ -44,9 +44,36 @@ On further review, it was determined that Collection was a special case of Objec
 
 Notification can safely be done asynchronously (including delaying it through a queue) but Modification by nature involve passing data back to the caller and thus must be synchronous.  Despite that difference the Working Group determined that the use cases were close enough to be considered in a single PSR.  The two different workflows however are represented by two different interfaces, `MessageNotifierInterface` (for Messages) and `TaskProcessorInterface` (for Tasks).
 
-### 4.2 Immutable events
+### 4.2 Intended use cases
 
-Initially the Working Group wished to define all Events as immutable message objects, similar to PSR-7.  However, that proved problematic in the Processor case.  In both of those cases Listeners needed a way to return data to the caller.  In concept there were three possible avenues:
+Despite their similarities, the Message and Task pipelines have two distinct workflows and use cases.
+
+| Messages                      | Tasks
+|-------------------------------|-------------------------------
+| Must be immutable             | May me mutable
+| Must be serializable          | May be serializable
+| May be delayed                | Must be processed immediately
+| Listener order not guaranteed | Listener order is guaranteed
+| All listeners will fire       | Listeners may short-circuit the pipeline (if the Stoppable interface is implemented)
+| One-way communication only    | May be one-way or two-way communication
+
+As a general guideline, a Message Notifier is appropriate when:
+
+* The Emitter does not care what responses are taken to an event.
+* The common case would be that only a single listener is registered, that is, when the Message Notifier is being used as a Command Bus.
+* The Emitter anticipates listeners to be expensive and therefore likely to be deferred for later processing.
+
+A Task Processor is appropriate when:
+
+* The event is being used as a "hook" or "pointcut" to extend or modify the Emitter's behavior.
+* The Emitter wishes to allow listeners to interact with each other (through mutating the event).
+* NEED A 3RD EXAMPLE HERE.
```

Comment:
How about: "The emitter chooses to delegate when processing is complete to the listeners."